pkg/controllers/externaloidc/generation/common.go

```
@@ -0,0 +1,235 @@
+	return fmt.Errorf("claims.email_verified must be used in claimMappings.username.expression or claimMappings.extra[*].valueExpression or claimValidationRules[*].expression when claims.email is used in claimMappings.username.expression")
+}
+
+// usesEmailClaim, usesEmailVerifiedClaim, anyUsesEmailVerifiedClaim, hasSelectExp,
+// isIdentOperand, and isConstField are copied from the upstream Kubernetes apiserver
+// CEL validation logic introduced in https://github.com/kubernetes/kubernetes/pull/123737 (commit 121607e):
+// https://github.com/kubernetes/kubernetes/blob/bfb362c57578518bed8e08a56a7318bab9b57429/staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go#L443
+func usesEmailClaim(ast *celgo.Ast) bool {
+	if ast == nil {
+		return false
+	}
+	return hasSelectExp(ast.Expr(), "claims", "email")
```

golangci-lint (2.12.2)

[error] SA1019: ast.Expr is deprecated: prefer cel.AstToCheckedExpr() or cel.AstToParsedExpr() and call GetExpr() the result instead. 

(staticcheck)